pkg/controller/elasticsearch/nodespec/readiness_probe.go

```diff
@@ -58,3 +57,23 @@ else
 	exit 1
 fi
 `
+
+const PreStopHookScriptConfigKey = "pre-stop-hook-script.sh"
+const PreStopHookScript = `#!/usr/bin/env bash
+
+# This script will wait for up to $MAX_WAIT_SECONDS for $POD_IP to disappear from DNS record,
+# then it will wait additional $ADDN_WAIT_SECONDS and exit. This slows down the process shutdown
+# and allows to make changes to the pool gracefully, without blackholing traffic when DNS
+# contains IP that is already inactive. Assumes $SERVICE_NAME and $POD_IP env variables are defined.
+
+MAX_WAIT_SECONDS=20 # max time to wait for pods IP to disappear from DNS
+ADDN_WAIT_SECONDS=1 # additional wait, allows queries to successfully use IP from old DNS entry
```

Comment:
Not sure I understand "allows queries to successfully use IP from old DNS entry", could you explain?

Reply:
Sure! With `ADDN_WAIT_SECONDS` set to 0 we would immediately exit when DNS record is right. But every query that resolved that name just before DNS record changed has the old one. This gives us one second for those clients to hit us (as it causes the IP from the old DNS entry to be correct for a little bit longer). I think this covers most of the cases with no keep alives and helps a lot with consistency during e2e tests.

Reply:
I'll reword it.